README.md

```diff
@@ -1,2 +1,242 @@
-# release-notes-presence-check
-A GH action for validating presence of release notes.
+# Release Notes Presence Checker
+
+A GH action for validating the presence of release notes in pull requests.
+
+## Motivation
+
+This action is designed to help maintainers and contributors ensure that release notes are present in pull requests. It can be used to prevent common issues such as:
+- Missing release notes
+- Incomplete release notes
+- Incorrect release note formats
+- Release notes in the wrong location
+
+## Requirements
+- **GitHub Token**: A GitHub token with permission to fetch repository data such as Issues and Pull Requests.
+- **Python 3.11+**: Ensure you have Python 3.11 installed on your system.
+
+## Inputs
+
+### `GITHUB_TOKEN`
+- **Description**: Your GitHub token for authentication. Store it as a secret and reference it in the workflow file as secrets.GITHUB_TOKEN.
+- **Required**: Yes
+
+### `pr-number`
+- **Description**: The pull request number to check for release notes. Example: `1`.
+- **Required**: Yes
+
+### `github-repository`
+- **Description**: The GitHub repository to check for version tags. Example: `AbsaOSS/release-notes-presence-check`.
+- **Required**: Yes
+
+### `location`
+- **Description**: The location of the release notes in the pull request. Example: `body`.
+- **Required**: No
+
+### `title`
+- **Description**: The title of the release notes in the pull request. Example without regex: `Release Notes:`, with regex: `[Rr]elease [Nn]otes:`.
+- **Required**: No
+
+### `skip-labels`
+- **Description**: The labels to skip the release notes check. Example: `skip-release-notes`.
+- **Required**: No
+
+### `fails-on-error`
+- **Description**: Whether the action should fail if an error occurs.
+- **Required**: No
+- **Default**: `true`
+
+## Outputs
+
+### `valid`
+- **Description**: Whether the release notes are present.
+- **Value**: `true` or `false`
+
+## Usage
+
+### Adding the Action to Your Workflow
+
+See the default action step definition:
+
+```yaml
+- name: Release Notes Presence Check
+  id: release_notes_presence_check
+  uses: AbsaOSS/release-notes-presence-check@v0.1.0
+  env:
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  
+  with:
+    github-repository: "{ org }/{ repo }"
+    pr-numer: 109
+    location: "body"
+    title: "[Rr]elease [Nn]otes:"
+    skip-labels: "skip-release-notes,no-release-notes"
+    fails-on-error: "false"
+  ```
+
+## Running Static Code Analysis
+
+This project uses Pylint tool for static code analysis. Pylint analyses your code without actually running it. It checks for errors, enforces, coding standards, looks for code smells etc.
+
+Pylint displays a global evaluation score for the code, rated out of a maximum score of 10.0. We are aiming to keep our code quality high above the score 9.5.
+
+### Set Up Python Environment
+```shell
+python3 -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+
+# run your commands
+
+deactivate
```

Comment:
I am not sure about the `#run your commands` and `deactivate`. Since the same block of code is below for black as well but without these two rows.

Reply:
You are right. It is not practical. Removed in commit TBD